server/src/servershot.js

```diff
@@ -542,6 +543,9 @@ Shot.getShotsForDevice = function(backend, deviceId, accountId, searchQuery, pag
   }).then(() => {
     const offset = (pageNumber - 1) * SHOTS_PER_PAGE;
     let sql, args, idNums;
+    if (!deviceIds.length) {
+      return [];
+    }
```

Comment:
@ianb Issue here is safeguard check is missing before query on Line 567 and 552. All other places where we execute sqls needing data.deviceid IN checks we have this safeguard code